hhs_oauth_server/middleware.py

```diff
@@ -0,0 +1,11 @@
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework.exceptions import APIException
+from django.http import HttpResponse
+
+
+# Mixin for compatibility with Django <1.10
+class HandleAPIExceptionMiddleware(MiddlewareMixin):
+    def process_exception(self, request, exception):
+        if isinstance(exception, APIException):
```

Comment:
While we're doing this, it would be good to add a generic exception catcher - we're currently returning an HTML 500 page on unknown exceptions.

Actually, I'm not really sure how this fits here, just because we could be looking to return JSON and we could be looking to return HTML.

Reply:
Django already has a generic exception catcher that we probably don't want to completely override. However there might be options to change what is returned from it.